packages/vp-message-handler/src/message-handler.ts

```diff
@@ -0,0 +1,35 @@
+import { IAgentContext, IResolver } from '@veramo/core'
+import { AbstractMessageHandler, Message } from '@veramo/message-handler'
+import Debug from 'debug'
+import { CredentialMapper, OriginalVerifiablePresentation } from '@sphereon/ssi-types'
+
+const debug = Debug('sphereon:vp-message-handler:message-handler')
+
+export type IContext = IAgentContext<IResolver>
+
+/**
+ * A plugin for {@link @sphereon/ssi-sdk-vp-message-handler#VpMessageHandler} that unpacks & uniforms a VC/VP
+ * @public
+ */
+export class VpMessageHandler extends AbstractMessageHandler {
+  async handle(message: Message, context: IContext): Promise<Message> {
+    if (message.raw) {
+      try {
+        const parsed = JSON.parse(message.raw)
+        if (parsed) {
+          if (parsed['jwt_vp']) {
+            message.raw = parsed['jwt_vp']
+            const jwtDecodedVp = CredentialMapper.toWrappedVerifiablePresentation(message.raw as OriginalVerifiablePresentation).decoded
+            if (jwtDecodedVp.aud) {
+              message.addMetaData({ type: 'JWT' })
```

Comment:
Is there are reason why we are not passing the wrapped VP in the message from this point on? We typically need to know the content if the VCs/VPs anyway

Reply:
At this point I am only unwrapping the raw credential from the message. toWrappedVerifiablePresentation is only done to mark it as JWT credential in the Veramo message.

The SharePoint agent module will decode it again to get the field info for the mappings. The original JWT will by the file data as it still contains the signature.

I can put the decoded version as message.data, it will save double work, I will try taht now. The "toUniformPresentation" call I would like to keep in the SharePoint module because now I need to call it with new option "repairMissingVpContext"

Reply:
done